common/scala/src/main/scala/whisk/core/database/CouchDbRestStore.scala

```diff
@@ -110,6 +134,39 @@ class CouchDbRestStore[DocumentAbstraction <: DocumentSerializer](
         transid.failed(this, start, s"[PUT] '$dbName' internal error, failure: '${failure.getMessage}'", ErrorLevel))
   }
 
+  private def put(ds: Seq[JsObject])(
+    implicit transid: TransactionId): Future[Seq[Either[ArtifactStoreException, DocInfo]]] = {
+    val count = ds.size
+    val start = transid.started(this, LoggingMarkers.DATABASE_BULK_SAVE, s"'$dbName' saving $count documents")
+
+    val f = client.putDocs(ds).map {
+      _ match {
+        case Right(response) =>
+          transid.finished(this, start, s"'$dbName' completed $count documents")
+
+          response.convertTo[Seq[BulkEntityResult]].map { singleResult =>
+            singleResult.error
+              .map {
+                case "conflict" => Left(DocumentConflictException("conflict on 'bulk_put'"))
```

Comment:
If there is one conflicting entity in the bulk request do all the entities fail?

Reply:
Nope, it's only a specific entity that fails. Behavior then is just like with any single request today.

Reply:
No, all other entries will have no errors. So there is no need to retry, if there is only one document with a conflict.